hooks/useClickOutside.tsx

```diff
@@ -1,18 +1,17 @@
 import { useEffect, useRef } from "react"
 
 interface IClickOutsideHook {
-  id: string
   shouldRegister: boolean
   onOutsideClick: () => any
 }
 
-function useClickOutside({ id, shouldRegister, onOutsideClick }: IClickOutsideHook) {
-  const ref = useRef<HTMLDivElement | null>(null)
+function useClickOutside({ shouldRegister, onOutsideClick }: IClickOutsideHook) {
+  const ref = useRef<HTMLDivElement | any | null>(null)
 
   const handleMouseClick = async (e: MouseEvent) => {
-    const node = e.target as any
+    const node = e.target as Node
 
-    if (id !== node?.id) return
+    if (ref && ref.current && ref.current.contains(node)) return
```

Comment:
Can't use optional chaining? `ref?.current?.contains(node)`

Reply:
Good catch. 🙏
I would highly prefer suggestions for shorter code with the same meaning. ⚡️
I have just fixed that line.